### PR TITLE
Support pre release

### PR DESCRIPTION
**Pull request by ahelal**
Thank you for your awesome work.
This PR add support for github releases. It use same approach as draft releases. Please note that I did not add tests. My go skills are still basic.

### CODE REVIEW

File: check_command.go [hunk 1]
```diff
@@ -67,7 +71,7 @@ func (c *CheckCommand) Run(request CheckRequest) ([]Version, error) {
 
 	for _, release := range filteredReleases {
 		if !upToLatest {
-			if *release.Draft {
+			if *release.Draft || *release.Prerelease {
```
Comment (shinji62) [inline check_command.go]: I think you should use CamelCase everywhere .

Comment (ahelal) [inline check_command.go]: That was my initial thought, but https://github.com/google/go-github/blob/master/github/repos_releases.go#L27 uses ```Prerelease``` so all ```*release.Prerelease``` will fail.
As for the rest I followed the  the same style as draft. let me know if should still remove them.

**Comment by vito** [comment]
I think this also has an interesting side-effect of having the resource ignore pre-releases by default. That makes sense to me, just wanted to make note of it.

It also treats them as an exclusive thing, though, i.e. a resource configured for pre-releases will not include final-releases. Which could also be confusing as the pre-release may eventually turn into a final release. I wonder if the behavior for turning it on in `source` should instead be to *include* pre-releases? That would also allow people to explicitly opt-in to them, and live on the bleeding edge.

**Comment by mainephd** [comment]
I was going to submit a pull request for this very thing until I saw this active pull request. The way we envisioned it would work was to have a `pre-release` resource configured along with a `release` resource configured. Each one triggering a different job in our pipeline. The job triggered by the `pre-release` would publish to a development area for testing. Once the testing was successfull we would mark the tag as `release`. Then `release` resource would ttrigger a job publish to production/acceptance area. Hopefully that makes sense.

**Comment by ahelal** [comment]
@vito It would be probably beyond my skills to adapt such functionality. 
But I would see the befit of such approach 
release_type by default both final and pre release if you define a release_type **final** or **pre** it will be restricted.

I am not sure how you want to proceed.

**Comment by ahelal** [comment]
So I gave it a go. Now it supports both uses cases.

**Comment by aturnerbulldawg** [comment]
Any ETA on this pull request getting merged in?  This has fairly significant consequences for us in order to run snapshot vs release pipelines.

**Comment by ahelal** [comment]
@vito That was a learning experience 👍 . So I added some tests. let me know your input. 
Thank you

File: out_command_test.go [hunk 2]
```diff
@@ -310,6 +310,77 @@ var _ = Describe("Out Command", func() {
...
+
+		Context("when release and prerelease are set", func() {
+			BeforeEach(func() {
+				bodyPath := filepath.Join(sourcesDir, "body")
+				file(bodyPath, "this is a great release")
+				request.Source.Release = true
+				request.Source.PreRelease = true
+			})
+
+			It("creates a release on GitHub in draft mode", func() {
```
Comment (vito) [inline out_command_test.go]: This should say "creates a final release on GitHub" or something

File: out_command_test.go [hunk 3]
```diff
@@ -310,6 +310,77 @@ var _ = Describe("Out Command", func() {
...
 
+		Context("when prerelease are set and realses are not", func() {
+			BeforeEach(func() {
+				bodyPath := filepath.Join(sourcesDir, "body")
+				file(bodyPath, "this is a great release")
+				request.Source.Release = false
+				request.Source.PreRelease = true
+			})
+
+			It("creates a release on GitHub in draft mode", func() {
```
Comment (vito) [inline out_command_test.go]: This should say "creates a non-draft pre-release on GitHub" or something

**Comment by ahelal** [comment]
third time's a charm.

File: out_command_test.go [hunk 4]
```diff
@@ -310,7 +310,7 @@ var _ = Describe("Out Command", func() {
 			Ω(*release.Draft).Should(Equal(false))
 		})
 
-		Context("when prerelease are set and realses are not", func() {
+		Context("creates a non-draft pre-release in Github", func() {
```
Comment (vito) [inline out_command_test.go]: misfire :) this was OK, it was the `It` below that was wrong

File: out_command_test.go [hunk 5]
```diff
@@ -346,7 +346,7 @@ var _ = Describe("Out Command", func() {
 			})
 		})
 
-		Context("when release and prerelease are set", func() {
+		Context("creates a final release in Github", func() {
```
Comment (vito) [inline out_command_test.go]: same as above

**Comment by vito** [comment]
fourth? :P

**Comment by ahelal** [comment]
hmm no more late night commits for me

**Comment by vito** [comment]
Looks good, thanks a bunch!

**Pull request merged**